frontend/src/app/testQueue/QueueItem.test.tsx

```diff
@@ -561,6 +561,59 @@ describe("QueueItem", () => {
     });
   });
 
+  it("highlights the test card where the validation failure occurs", async () => {
+    render(
+      <MockedProvider mocks={mocks}>
+        <Provider store={store}>
+          <QueueItem
+            internalId={testProps.internalId}
+            patient={testProps.patient}
+            askOnEntry={testProps.askOnEntry}
+            selectedDeviceId={testProps.selectedDeviceId}
+            selectedDeviceTestLength={testProps.selectedDeviceTestLength}
+            selectedDeviceSpecimenTypeId={
+              testProps.selectedDeviceSpecimenTypeId
+            }
+            deviceSpecimenTypes={testProps.deviceSpecimenTypes}
+            selectedTestResult={testProps.selectedTestResult}
+            devices={testProps.devices}
+            refetchQueue={testProps.refetchQueue}
+            facilityId={testProps.facilityId}
+            dateTestedProp={testProps.dateTestedProp}
+            facilityName="Foo facility"
+          />
+        </Provider>
+      </MockedProvider>
+    );
+
+    // Enter an invalid (future) date on the first test card
+    const toggle = await screen.findByLabelText("Current date/time");
+    userEvent.click(toggle);
+    const dateInput = await screen.findByTestId("test-date");
+    expect(dateInput).toBeInTheDocument();
+    userEvent.type(dateInput, moment().add(5, "days").format("YYYY-MM-DD"));
+    dateInput.blur();
+
+    // Select result
+    userEvent.click(
+      await screen.findByLabelText("Inconclusive", {
+        exact: false,
+      })
+    );
+
+    await new Promise((resolve) => setTimeout(resolve, 600));
```

Comment:
Is there any way to avoid doing this? Can you instead use `waitFor()` to wait for the submit button to be enabled?

Reply:
I can do just that! See [f7cdf10](https://github.com/CDCgov/prime-simplereport/pull/3279/commits/f7cdf1088c247b94544087c67845b247b3249dec).